nuxeo-core/nuxeo-core-event/src/test/java/org/nuxeo/ecm/core/work/StreamWorkManagerTest.java

```diff
@@ -145,7 +152,7 @@ public void testWorkIdempotentConcurrent() throws InterruptedException {
         service.schedule(work2);
         // we don't know if work1 and work2 are executed on the same thread
         // but we assume that the max duration is work1 + work2 because there is only one invocation of each
-        assertTrue(service.awaitCompletion(2500, TimeUnit.MILLISECONDS));
+        assertTrue(service.awaitCompletion(getDurationMillis() * 3, TimeUnit.MILLISECONDS));
```

Comment:
in theory it takes 2s, if it takes 3s it does not work because it means that work1 is executed twice.
The timeout must be lower than 3s try something like "getDurationMillis() * 3 - 200"